crates/biome_js_analyze/src/lint/nursery/no_img_element.rs

```diff
@@ -0,0 +1,137 @@
+use biome_analyze::RuleSourceKind;
+use biome_analyze::{
+    context::RuleContext, declare_lint_rule, Ast, Rule, RuleDiagnostic, RuleSource,
+};
+use biome_console::markup;
+use biome_js_syntax::{
+    JsSyntaxToken, JsxAttributeList, JsxChildList, JsxElement, JsxOpeningElement,
+    JsxSelfClosingElement,
+};
+use biome_rowan::{declare_node_union, AstNode, AstNodeList, TextRange};
+
+declare_lint_rule! {
+    /// Prevent usage of `<img>` element in a Next.js project.
+    ///
+    /// Using the `<img>` element can result in slower Largest Contentful Paint (LCP)
+    /// and higher bandwidth usage, as it lacks the optimizations provided by the `<Image />`
+    /// component from `next/image`. Next.js's `<Image />` automatically optimizes images
+    /// by serving responsive sizes and using modern formats, improving performance and reducing bandwidth.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <img alt="Foo" />
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <div>
+    ///   <img alt="Foo" />
+    /// </div>
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```jsx
+    /// <img />
+    /// ```
+    ///
+    /// ```jsx
+    /// <Image src="https://example.com/hero.jpg" />
+    /// ```
+    ///
+    /// ```jsx
+    /// <picture>
+    ///   <source srcSet="https://example.com/hero.avif" type="image/avif" />
+    ///   <source srcSet="https://example.com/hero.webp" type="image/webp" />
+    ///   <img src="https://example.com/hero.jpg" />
+    /// </picture>
+    /// ```
+    ///
+    pub NoImgElement {
+        version: "next",
+        name: "noImgElement",
+        language: "jsx",
+        sources: &[RuleSource::EslintNext("no-img-element")],
+        source_kind: RuleSourceKind::SameLogic,
+        recommended: false,
+    }
+}
+
+declare_node_union! {
+    pub NoImgElementQuery = JsxOpeningElement | JsxSelfClosingElement
```

Comment:
Internally, we have a naming convention to call `Any*` all the unions of nodes. You can check other usages to understand how we do it